code/e2e-tests/json-files.spec.ts

```diff
@@ -0,0 +1,42 @@
+import { test, expect } from '@playwright/test';
+import process from 'process';
+
+const storybookUrl = process.env.STORYBOOK_URL || 'http://localhost:8001';
+
+test.describe('JSON files', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto(storybookUrl);
+  });
+
+  test('should have index.json', async ({ page }) => {
+    // eslint-disable-next-line jest/valid-expect-in-promise
+    const json = await page.evaluate(() => fetch('/index.json').then((res) => res.json()));
+
+    expect(json).toEqual({
+      v: expect.any(Number),
+      entries: expect.objectContaining({
+        'example-button--primary': {
+          id: 'example-button--primary',
+          importPath: expect.stringMatching(/\/Button\.stories\.[tj]sx$/),
+          name: 'Primary',
+          title: 'Example/Button',
+          type: 'story',
+        },
+      }),
+    });
+  });
+
+  test('should have modules.json', async ({ page }) => {
```

Comment:
It would be good to test

(a) there is another file called "something like `Button`"[1] that is there due to `Button.stories.xyz`.
(b) `Button.stories.xyz` is detected as a CSF file.

[1] The file name of button can be quite a few things (e.g. `Button.component.ts`, `Button.svelte` etc) but it'll always have `Button` in it.